cbmc/patches/0001-Simulates-background-thread-behavior-in-IotMqtt_Connect.patch

```diff
@@ -0,0 +1,32 @@
+From 9119399f0e3bfd896cba40e924d52a919801d4c1 Mon Sep 17 00:00:00 2001
+From: "Felipe R. Monteiro" <felisous@amazon.com>
+Date: Thu, 9 Apr 2020 18:34:33 +0000
+Subject: [PATCH] Simulates background thread behavior in IotMqtt_Connect
+
+Signed-off-by: Felipe R. Monteiro <felisous@amazon.com>
+---
+ libraries/standard/mqtt/src/iot_mqtt_api.c | 8 ++++++++
+ 1 file changed, 8 insertions(+)
+
+diff --git a/libraries/standard/mqtt/src/iot_mqtt_api.c b/libraries/standard/mqtt/src/iot_mqtt_api.c
+index 1e55cb5..631a3e1 100644
+--- a/libraries/standard/mqtt/src/iot_mqtt_api.c
++++ b/libraries/standard/mqtt/src/iot_mqtt_api.c
+@@ -1005,6 +1005,14 @@ static IotMqttError_t _sendConnectRequest( _mqttOperation_t * pOperation,
+     /* Wait for the CONNECT operation to complete, i.e. wait for CONNACK. */
+     status = IotMqtt_Wait( pOperation, timeoutMs );
+ 
++    /* There is a background thread that's part of the network and changes
++     * the status from IOT_MQTT_STATUS_PENDING to IOT_MQTT_SUCCESS. This
```

Comment:
This should also be protected by ifdefs.  The point is that we don't want this kind of thing to escape into the wild by accident, like accidentally checking into the repo the patched files generated during debugging..  Patches should always be protected.